sdk/core/azure-core/src/main/java/com/azure/android/core/util/logging/AndroidClientLogger.java

```diff
@@ -0,0 +1,163 @@
+
+    /**
+     * Indicates that log level is at the debug level.
+     */
+    public static final int DEBUG = 1;
+
+    /**
+     * Indicates that log level is at the informational level.
+     */
+    public static final int INFO = 2;
+
+    /**
+     * Indicates that log level is at the warning level.
+     */
+    public static final int WARNING = 3;
+
+    /**
+     * Indicates that log level is at the error level.
+     */
+    public static final int ERROR = 4;
+
+    /**
+     * Indicates that no log level is set.
+     */
+    public static final int NOT_SET = 5;
+
+    private final String tag;
+    private int logLevel;
+
+    /**
+     * Retrieves a logger for the passed class using the {@link LoggerFactory}.
+     *
+     * @param clazz Class creating the logger.
+     */
+    public AndroidClientLogger(Class<?> clazz) {
+        this(clazz.getName());
+    }
+
+    /**
+     * Retrieves a logger for the passed class name using the {@link LoggerFactory}.
+     *
+     * @param tag Class name creating the logger.
+     */
+    public AndroidClientLogger(String tag) {
+        this.tag = tag;
+    }
+
+    /**
+     * Returns this logger's log level.
+     *
+     * @return The log level.
+     */
+    @LogLevel
+    public int getLogLevel() {
+        return logLevel;
+    }
+
+    /**
+     * Sets the log level for this logger.
+     */
+    public void setLogLevel(@LogLevel int logLevel) {
+        this.logLevel = logLevel;
+    }
+
+    public void debug(String message) {
+        if (canLogAtLevel(DEBUG)) {
+            Log.d(tag, message);
+        }
+    }
+
+    public void debug(String message, Throwable throwable) {
+        if (canLogAtLevel(DEBUG)) {
+            Log.d(tag, message, throwable);
+        }
+    }
+
+    public void info(String message) {
+        if (canLogAtLevel(INFO)) {
+            Log.i(tag, message);
+        }
+    }
+
+    public void info(String message, Throwable throwable) {
+        if (canLogAtLevel(INFO)) {
+            Log.i(tag, message, throwable);
+        }
+    }
+
+    public void warning(String message) {
+        if (canLogAtLevel(WARNING)) {
+            Log.w(tag, message);
+        }
+    }
+
+    public void warning(String message, Throwable throwable) {
+        if (canLogAtLevel(WARNING)) {
+            Log.w(tag, message, throwable);
+        }
+    }
+
+    public void error(String message) {
+        if (canLogAtLevel(ERROR)) {
+            Log.e(tag, message);
+        }
+    }
+
+    public void error(String message, Throwable throwable) {
+        if (canLogAtLevel(ERROR)) {
+            Log.e(tag, message, throwable);
+        }
+    }
+
+    /**
+     * Determines if the logger supports logging at the given log level.
+     *
+     * @param logLevel Logging level to validate.
+     * @return Flag indicating if the environment and logger are configured to support logging at the given log level.
+     */
+    private boolean canLogAtLevel(int logLevel) {
```

Comment:
I didn't do that since this is a private method and we control who calls it, but I think it doesn't hurt to use annotations to make explicitly clear what is expected in this method.

Reply:
Yep, at the very least it provides some nice intellisense when so annotated.